components/create_post/create_post.jsx

```diff
@@ -1128,6 +1145,9 @@ export default class CreatePost extends React.Component {
                                 onChange={this.handleChange}
                                 onKeyPress={this.postMsgKeyPress}
                                 onKeyDown={this.handleKeyDown}
+                                onCompositionStart={this.handleCompositionStart}
+                                onCompositionUpdate={this.handleCompositionUpdate}
+                                onCompositionEnd={this.handleCompositionEnd}
```

Comment:
Looks repetitive to me.  Can we just pass one function?  Maybe something like,
```suggestion
                                onComposition={this.emitTypingEvent}
```

Then on `SuggestionBox` component, access it like below for `handleCompositionStart|Update|End` functions,
```javascript
if (this.props.onComposition) {
    this.props.onComposition();
}
```

Reply:
When apply your suggestion, below error message was printed in my browser console (and full stack trace is [here](https://gist.github.com/maruTA-bis5/aed008da0f5a432872d0ca514dba20d2)).
> Warning: Unknown event handler property `onComposition`. It will be ignored.

It may caused by `SuggestionBox` component passes `{..props}` to `QuickInput` and `props` contains `onComposition`.

Do you know how exclude `onComposition` from `{..props}`?

Reply:
Sorry for late reply.  I think it can be done something like:
```diff
diff --git a/components/suggestion/suggestion_box.jsx b/components/suggestion/suggestion_box.jsx
index bfa89517..6a46cffd 100644
--- a/components/suggestion/suggestion_box.jsx
+++ b/components/suggestion/suggestion_box.jsx
@@ -80,9 +80,7 @@ export default class SuggestionBox extends React.Component {
          * Function called when a key is pressed and the input box is in focus
          */
         onKeyDown: PropTypes.func,
-        onCompositionStart: PropTypes.func,
-        onCompositionUpdate: PropTypes.func,
-        onCompositionEnd: PropTypes.func,
+        onComposition: PropTypes.func,
 
         /**
          * Function called when an item is selected
@@ -302,8 +300,8 @@ export default class SuggestionBox extends React.Component {
         this.pretext = pretext;
-        if (this.props.onCompositionUpdate) {
-            this.props.onCompositionUpdate();
+        if (this.props.onComposition) {
+            this.props.onComposition();
         }
     }
 
     handleCompositionEnd = () => {
         this.composing = false;
-        if (this.props.onCompositionEnd) {
-            this.props.onCompositionEnd();
+        if (this.props.onComposition) {
+            this.props.onComposition();
         }
     }
 
@@ -614,6 +612,7 @@ export default class SuggestionBox extends React.Component {
         // Don't pass props used by SuggestionBox
         Reflect.deleteProperty(props, 'providers');
         Reflect.deleteProperty(props, 'onChange'); // We use onInput instead of onChange on the actual input
+        Reflect.deleteProperty(props, 'onComposition');
         Reflect.deleteProperty(props, 'onItemSelected');
         Reflect.deleteProperty(props, 'completeOnTab');
         Reflect.deleteProperty(props, 'isRHS');
```

Reply:
I missed `Reflect.deleteProperty`!
Thank you Saturnino. I'll try that!